internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,444 @@
+	SaslMechanisms   string `yaml:"sasl.mechanisms"`
+	SaslPassword     string `yaml:"sasl.password"`
+	SaslUsername     string `yaml:"sasl.username"`
+	SecurityProtocol string `yaml:"security.protocol"`
+}
+
+type ConfluentBroker struct {
+	Type     string                  `yaml:"type"`
+	XConfigs ConfluentBrokerXConfigs `yaml:"x-configs"`
+}
+
+type ConfluentSRXConfigs struct {
+	BasicAuthUserInfo string `yaml:"basic.auth.user.info"`
+}
+type ConfluentSchemaRegistry struct {
+	Type     string              `yaml:"type"`
+	XConfigs ConfluentSRXConfigs `yaml:"x-configs"`
+}
+
+func newImportCommand(prerunner pcmd.PreRunner) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "import <spec.yaml>",
+		Short: "Adds topics, schemas and tags from the yaml file.",
+		Args:  cobra.ExactArgs(1),
+		Example: examples.BuildExampleString(
+			examples.Example{Text: "Import an asyncapi specification file to populate an existing cluster",
+				Code: "confluent asyncapi import spec.yaml",
+			},
+		),
+	}
+	c := &command{pcmd.NewAuthenticatedStateFlagCommand(cmd, prerunner), nil}
+	cmd.RunE = c.parse
+	cmd.Flags().Bool("overwrite", false, "Overwrite existing topics and schemas.")
+	cmd.Flags().String("kafka-api-key", "", "Kafka cluster API key.")
+	cmd.Flags().String("schema-registry-api-key", "", "API key for Schema Registry.")
+	cmd.Flags().String("schema-registry-api-secret", "", "API secret for Schema Registry.")
+
+	return cmd
+}
+
+func getFlagsImport(cmd *cobra.Command) (*flagsImport, error) {
+	overwrite, err := cmd.Flags().GetBool("overwrite")
+	if err != nil {
+		return nil, err
+	}
+	kafkaApiKey, err := cmd.Flags().GetString("kafka-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiKey, err := cmd.Flags().GetString("schema-registry-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiSecret, err := cmd.Flags().GetString("schema-registry-api-secret")
+	if err != nil {
+		return nil, err
+	}
+
+	flags := &flagsImport{
+		overwrite:               overwrite,
+		kafkaApiKey:             kafkaApiKey,
+		schemaRegistryApiKey:    schemaRegistryApiKey,
+		schemaRegistryApiSecret: schemaRegistryApiSecret,
+	}
+	return flags, nil
+}
+
+func (c *command) parse(cmd *cobra.Command, args []string) error {
+	spec, err := fileToStruct(args[0])
+	if err != nil {
+		return err
+	}
+	// Get flags
+	flagsImp, err := getFlagsImport(cmd)
+	if err != nil {
+		return err
+	}
+	// Getting Kafka Cluster & SR
+	flags := createExportFlagsFromImportFlags(flagsImp)
+	details, err := c.getAccountDetails(&flags)
+	if err != nil {
+		return err
+	}
+	for topicName, topicDetails := range spec.Channels {
+		err = c.addChannelToCluster(details, spec, topicName, topicDetails.TopicBinding.Kafka, false, flagsImp.overwrite)
+		if err != nil {
+			if err.Error() == parseErrorMessage {
+				log.CliLogger.Info(err)
+			} else {
+				log.CliLogger.Warn(err)
+			}
+		}
+	}
+	output.Printf("AsyncAPI specification imported from \"%s\"\n", args[0])
```

Comment:
It's the same specification. The input file is used to create topics, add schemas and tags. Printing the input file name should be enough instead of printing the entire specification.

Reply:
I think I understand things a little better now. Let's print messages for all of the resources that are added, but no need to include this message at the end. If the command exits without an error, we're done.
```suggestion
```

Reply:
This print statement was an ask from the Product Manager: [thread](https://confluent.slack.com/archives/C0438U051GS/p1678811225483179?thread_ts=1678780488.336289&cid=C0438U051GS)